packages/app/src/cli/services/dev/extension/bundler.ts

```diff
@@ -244,20 +246,49 @@ export async function setupFunctionWatcher({
 }: SetupFunctionWatcherOptions) {
   const {default: chokidar} = await import('chokidar')
 
-  outputDebug(`Starting watcher for function extension ${extension.devUUID}`, stdout)
-  const watchPaths = extension.watchPaths
-  if (!watchPaths) {
+  const rebuildAndRedeployWatchPaths = extension.watchPaths
+  const redeployWatchPaths: string[] = []
+
+  if (!rebuildAndRedeployWatchPaths) {
     outputWarn(
       `Function extension ${extension.localIdentifier} is missing the 'build.watch' setting, automatic builds are disabled.`,
       stdout,
     )
     return
   }
 
-  outputDebug(`Watching paths for function extension ${extension.localIdentifier}: ${watchPaths}`, stdout)
+  if (await fileExists(joinPath(extension.directory, 'locales'))) {
+    redeployWatchPaths.push(joinPath(extension.directory, 'locales', '**.json'))
+  }
+
+  outputDebug(
+    `Watching paths for function extension ${extension.localIdentifier}: ${rebuildAndRedeployWatchPaths.concat(
+      redeployWatchPaths,
+    )}`,
+    stdout,
+  )
```

Comment:
Nit: For the purposes of debugging, I wonder if it would be handy to know which paths trigger a rebuild and which trigger a redeploy.

Reply:
Updated the debug output message.